pandas/_libs/src/vendored/numpy/datetime/np_datetime.c

```diff
@@ -1070,5 +1070,8 @@ void pandas_timedelta_to_timedeltastruct(npy_timedelta td,
  */
 PyArray_DatetimeMetaData
 get_datetime_metadata_from_dtype(PyArray_Descr *dtype) {
-  return (((PyArray_DatetimeDTypeMetaData *)dtype->c_metadata)->meta);
+  #if NPY_ABI_VERSION < 0x02000000
+    #define PyDataType_C_METADATA(descr) ((descr)->c_metadata)
+  #endif
+  return (PyDataType_C_METADATA(descr)->meta);
```

Comment:
```suggestion
    #define PyDataType_C_METADATA(descr) ((dtype)->c_metadata)
  #endif
  return (PyDataType_C_METADATA(dtype)->meta);
```
Sorry, we mostly use `descr` for dtype instances inside NumPy so wrote the wrong thing.